### PR TITLE
[zuul] Update the tempest job to use test_operator

### PR DESCRIPTION
**Pull request by elfiesmelfie**
Using the tempest role from ci_framework ran the tempest container via podman on the controller.
test_operator uses the same image and runs tempest in a pod on the OCP cluster.

Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/659

### CODE REVIEW

**Comment by SeanMooney** [comment]
something broke in zuul but the job actually passed.

https://logserver.rdoproject.org/65/665/796bce2a61df96939f273e23905f29684659f55f/github-check/nova-operator-tempest-multinode/39fd880/controller/ci-framework-data/tests/test_operator/stestr_results.html

however the allowed and excluded list is not propagating so it only ran the default

**Comment by elfiesmelfie** [comment]
> however the allowed and excluded list is not propagating so it only ran the default

That's my bad, I forgot to update the names of the vars used to pass in the include_list and exclude_list, the update is pushed

**Comment by elfiesmelfie** [comment]
recheck

The Depends on link was not correct

**Comment by SeanMooney** [comment]
rdo-check

**Comment by SeanMooney** [comment]
check-github

**Comment by SeanMooney** [comment]
recheck there are a lot of rabbitmq and mysql connection errors in the failed tempest run whcih are unrelated to how we ran tempest so that could explain many of the test failures

File: .zuul.yaml [hunk 1]
```diff
@@ -118,62 +118,44 @@
...
-      # We don't have shared local storage
-        - tempest.api.compute.admin.test_live_migration.LiveMigrationTest.test_live_migration_with_trunk
-        - tempest.scenario.test_network_advanced_server_ops.TestNetworkAdvancedServerOps.test_server_connectivity_live_migration
-      # Need to check
-        - tempest.api.compute.admin.test_auto_allocate_network.AutoAllocateNetworkTest.test_server_multi_create_auto_allocate
-        - tempest.api.compute.servers.test_server_rescue.ServerStableDeviceRescueTestIDE
-      # Swift test failing with unauthorized errors
-        - tempest.scenario.test_object_storage
+      test_fw: test_operator
+      # NOTE(gibi): identity.v3_endpoint_type override is a WA to force the
```
Comment (SeanMooney) [inline .zuul.yaml:122]: we need to add 

cifmw_test_operator_timeout: 7200

the default is 3600 and i think we had i higher default previously

**Comment by gibizer** [comment]
Tempest logs are here now: https://review.rdoproject.org/zuul/build/b6577dec951b4bb3a7f746d47b5fae85/log/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/tempest-tests-6ksc4/logs/tempest-tests-tests-runner.log

The execution was interrupted by the job timeout. But I see a lot of failures in the executed tests that makes the execution slower e.g.:
```
{7} tempest.api.compute.security_groups.test_security_group_rules_negative.SecurityGroupRulesNegativeTestJSON.test_create_security_group_rule_with_invalid_port_range [67.534572s] ... FAILED
{6} tempest.api.compute.servers.test_delete_server.DeleteServersTestJSON.test_delete_server_while_in_building_state [98.352015s] ... FAILED
```
I think the main reason is to much parallelism. In the current podman based executor we run tempest in 4 processes
https://logserver.rdoproject.org/66/666/d9aa648cb7613bb124fd9d878fe97ab00d77664d/github-check/nova-operator-tempest-multinode/12ca46c/controller/ci-framework-data/tests/tempest/podman_tempest.log
but it seems the tempest operator uses 8.

**Comment by SeanMooney** [comment]
ya we likely need to drop it down to 3-4

ill check with emma in the morning and see if she wants us to take over this patch.

one downside to the test-operator is it only copies the full tempest logs to the logs folder if the tempest execution does not time out.

where it runes properly it provides the html report and the raw tempest logs in a seperate tests log dir but where waiting for the job to complete fails it does not collect those logs properly.

**Comment by lpiwowar** [comment]
> one downside to the test-operator is it only copies the full tempest logs to the logs folder if the tempest execution does not time out.
I'm going to take a look at this ^^. 

Also, about the concurrency. It would probably be good to drop it down to 4. I'm going to propose a patch. We've encountered similar issues related to it.

**Comment by lpiwowar** [comment]
Hi @elfiesmelfie, this should fix the timeout issue -> https://github.com/openstack-k8s-operators/ci-framework/pull/1112. 

I checked the logs and it seems that tempest pod finished the execution of the tempest tests but the role kept waiting [1].

[1] https://logserver.rdoproject.org/65/665/8a814593c6acbbf86b6ad3022bb565bb23d73247/github-check/nova-operator-tempest-multinode/66b37ab/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/tempest-tests-w26n7/logs/tempest-tests-tests-runner.log

**Comment by SeanMooney** [comment]
rdo-check

**Comment by SeanMooney** [comment]
i can see in the pod logs that nova and keystone were unable to connect to the db for a protracted period of time. i belive we have a tracker for a similar issue in other ci jobs so i think the test failures are directly a result of the db being inaccessable.

**Comment by SeanMooney** [comment]
much much better, only one test failed this time  test_mtu_sized_frames
and I'm not sure we configure tempest to run that properly as we have to take into account the lower mtu in ci.

**Comment by gibizer** [comment]
These are the lost test cases with this PR:
```
473,480d472
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_create_volume_with_invalid_size
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_create_volume_without_passing_size
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_create_volume_with_size_zero
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_delete_invalid_volume_id
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_delete_volume_without_passing_volume_id
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_get_volume_without_passing_volume_id
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_volume_delete_nonexistent_volume_id
< tempest.api.compute.volumes.test_volumes_negative.VolumesNegativeTest.test_volume_get_nonexistent_volume_id
491d482
< tempest.scenario.test_network_basic_ops.TestNetworkBasicOps.test_mtu_sized_frames
```
I'm OK to not running them here

**Review by gibizer — APPROVED** [review]

**Comment by SeanMooney** [comment]
check-rdo

**Pull request merged**